### ISSUE TITLE
NumPy < 1.9 on Python 2.7 missing gfortran library dependency

### ISSUE THREAD

**Issue by QuLogic** [ContinuumIO/anaconda-issues#445]
I guess something else in the stack depended on this library and so NumPy worked fine, but now no longer does.

```
$ for i in 5 6 7 8 9; do
    echo $i
    conda create -n np1${i} -y python=2.7 numpy=1.${i}\*
    source activate np1${i}
    python -c 'import numpy'
    source deactivate
  done

5
Package plan for installation in environment .../conda/envs/np15:

The following NEW packages will be INSTALLED:
    numpy:      1.5.1-py27_4 
    openssl:    1.0.1k-1     
    pip:        7.1.2-py27_0 
    python:     2.7.10-1     
    readline:   6.2-2        
    setuptools: 18.1-py27_0  
    sqlite:     3.8.4.1-1    
    tk:         8.5.18-0     
    wheel:      0.24.0-py27_0
    zlib:       1.2.8-0      

...

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File ".../envs/np15/lib/python2.7/site-packages/numpy/__init__.py", line 136, in <module>
    import add_newdocs
  File ".../envs/np15/lib/python2.7/site-packages/numpy/add_newdocs.py", line 9, in <module>
    from numpy.lib import add_newdoc
  File ".../envs/np15/lib/python2.7/site-packages/numpy/lib/__init__.py", line 13, in <module>
    from polynomial import *
  File ".../envs/np15/lib/python2.7/site-packages/numpy/lib/polynomial.py", line 17, in <module>
    from numpy.linalg import eigvals, lstsq
  File ".../envs/np15/lib/python2.7/site-packages/numpy/linalg/__init__.py", line 48, in <module>
    from linalg import *
  File ".../envs/np15/lib/python2.7/site-packages/numpy/linalg/linalg.py", line 23, in <module>
    from numpy.linalg import lapack_lite
ImportError: libgfortran.so.1: cannot open shared object file: No such file or directory

6
Package plan for installation in environment .../envs/np16:

The following NEW packages will be INSTALLED:
    numpy:      1.6.2-py27_4 
    openssl:    1.0.1k-1     
    pip:        7.1.2-py27_0 
    python:     2.7.10-1     
    readline:   6.2-2        
    setuptools: 18.1-py27_0  
    sqlite:     3.8.4.1-1    
    tk:         8.5.18-0     
    wheel:      0.24.0-py27_0
    zlib:       1.2.8-0      

...

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File ".../envs/np16/lib/python2.7/site-packages/numpy/__init__.py", line 137, in <module>
    import add_newdocs
  File ".../envs/np16/lib/python2.7/site-packages/numpy/add_newdocs.py", line 9, in <module>
    from numpy.lib import add_newdoc
  File ".../envs/np16/lib/python2.7/site-packages/numpy/lib/__init__.py", line 13, in <module>
    from polynomial import *
  File ".../envs/np16/lib/python2.7/site-packages/numpy/lib/polynomial.py", line 17, in <module>
    from numpy.linalg import eigvals, lstsq
  File ".../envs/np16/lib/python2.7/site-packages/numpy/linalg/__init__.py", line 48, in <module>
    from linalg import *
  File ".../envs/np16/lib/python2.7/site-packages/numpy/linalg/linalg.py", line 23, in <module>
    from numpy.linalg import lapack_lite
ImportError: libgfortran.so.1: cannot open shared object file: No such file or directory

7
Package plan for installation in environment .../envs/np17:

The following NEW packages will be INSTALLED:
    numpy:      1.7.1-py27_2 
    openssl:    1.0.1k-1     
    pip:        7.1.2-py27_0 
    python:     2.7.10-1     
    readline:   6.2-2        
    setuptools: 18.1-py27_0  
    sqlite:     3.8.4.1-1    
    tk:         8.5.18-0     
    wheel:      0.24.0-py27_0
    zlib:       1.2.8-0      

...

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File ".../envs/np17/lib/python2.7/site-packages/numpy/__init__.py", line 137, in <module>
    import add_newdocs
  File ".../envs/np17/lib/python2.7/site-packages/numpy/add_newdocs.py", line 9, in <module>
    from numpy.lib import add_newdoc
  File ".../envs/np17/lib/python2.7/site-packages/numpy/lib/__init__.py", line 13, in <module>
    from polynomial import *
  File ".../envs/np17/lib/python2.7/site-packages/numpy/lib/polynomial.py", line 17, in <module>
    from numpy.linalg import eigvals, lstsq, inv
  File ".../envs/np17/lib/python2.7/site-packages/numpy/linalg/__init__.py", line 48, in <module>
    from linalg import *
  File ".../envs/np17/lib/python2.7/site-packages/numpy/linalg/linalg.py", line 23, in <module>
    from numpy.linalg import lapack_lite
ImportError: libgfortran.so.1: cannot open shared object file: No such file or directory

8
Package plan for installation in environment .../conda/envs/np18:

The following NEW packages will be INSTALLED:
    numpy:      1.8.2-py27_0 
    openssl:    1.0.1k-1     
    pip:        7.1.2-py27_0 
    python:     2.7.10-1     
    readline:   6.2-2        
    setuptools: 18.1-py27_0  
    sqlite:     3.8.4.1-1    
    tk:         8.5.18-0     
    wheel:      0.24.0-py27_0
    zlib:       1.2.8-0      

...

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File ".../envs/np18/lib/python2.7/site-packages/numpy/__init__.py", line 153, in <module>
    from . import add_newdocs
  File ".../envs/np18/lib/python2.7/site-packages/numpy/add_newdocs.py", line 13, in <module>
    from numpy.lib import add_newdoc
  File ".../envs/np18/lib/python2.7/site-packages/numpy/lib/__init__.py", line 18, in <module>
    from .polynomial import *
  File ".../envs/np18/lib/python2.7/site-packages/numpy/lib/polynomial.py", line 19, in <module>
    from numpy.linalg import eigvals, lstsq, inv
  File ".../envs/np18/lib/python2.7/site-packages/numpy/linalg/__init__.py", line 50, in <module>
    from .linalg import *
  File ".../envs/np18/lib/python2.7/site-packages/numpy/linalg/linalg.py", line 29, in <module>
    from numpy.linalg import lapack_lite, _umath_linalg
ImportError: libgfortran.so.1: cannot open shared object file: No such file or directory

9
Package plan for installation in environment .../conda/envs/np19:

The following NEW packages will be INSTALLED:
    libgfortran: 1.0-0        
    numpy:       1.9.2-py27_1 
    openblas:    0.2.14-1     
    openssl:     1.0.1k-1     
    pip:         7.1.2-py27_0 
    python:      2.7.10-1     
    readline:    6.2-2        
    setuptools:  18.1-py27_0  
    sqlite:      3.8.4.1-1    
    tk:          8.5.18-0     
    wheel:       0.24.0-py27_0
    zlib:        1.2.8-0      

...

< No Message >
```
Note: I trimmed some of extraneous downloading/linking/activating messages (that don't change) to keep this shorter.

**Comment by QuLogic** [comment]
PS, running `conda install libgfortran` in any of those environments above fixes the NumPy import.

**Comment by pelson** [comment]
There were a lot of numpy builds yesterday which were added: https://anaconda.org/anaconda/numpy/files

I suspect the recipe has been updated for some purpose.
Ping @asmeurer & @ilanschnell - this is pretty major - you can't create an environment with numpy in it without knowing specifying, or finding something else which depends upon, libgfortran.

**Comment by ilanschnell** [comment]
Good catch.  So we need to add `libgfortran` as a dependency of the older numpy versions on Linux.  I hope I'll have a fix shortly/

**Comment by ilanschnell** [comment]
So, I've fixed it for numpy 1.8, I'll also try to do the fix for 1.7, 1.6 and 1.5

**Comment by ilanschnell** [comment]
Should be fixed now for all previous numpy versions.

**Comment by pelson** [comment]
Thanks @ilanschnell!

**Comment by markunsworth** [comment]
We're also seeing same issue in scipy.
```
>>> import scipy.stats as sps
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/vol/jenkins/workspace/model-generation/env/lib/python2.7/site-packages/scipy/stats/__init__.py", line 321, in <module>
    from .stats import *
  File "/vol/jenkins/workspace/model-generation/env/lib/python2.7/site-packages/scipy/stats/stats.py", line 180, in <module>
    import scipy.special as special
  File "/vol/jenkins/workspace/model-generation/env/lib/python2.7/site-packages/scipy/special/__init__.py", line 601, in <module>
    from ._ufuncs import *
ImportError: libgfortran.so.1: cannot open shared object file: No such file or directory
```

**Comment by nouiz** [comment]
Just to tell that the problem of the missing dependency is still there. The fix for scipy 0.14 changed to:

    conda install libgfortran==1
